src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/DownloadManager.java

```diff
@@ -235,6 +236,15 @@ public Path download(
     } else if (repositoryCache.isEnabled()) {
       String newSha256 = repositoryCache.put(destination, KeyType.SHA256, canonicalId);
       eventHandler.handle(Event.info("SHA256 (" + rewrittenUrls.get(0) + ") = " + newSha256));
+      
+      if(!sha256.isEmpty()) {
+        try {
+          Checksum.stringComparison(sha256, newSha256);
+        }
+        catch (Checksum.ICExceptionExtended e) {
+          throw new IOException(e.getMessage());
```

Comment:
If you're throwing an IOException with the message of the thrown exception, then why not just make ICExceptionExtended a subclass of IOException, and not catch it here at all?